src/hera/workflows/_mixins.py

```diff
@@ -1085,27 +1089,31 @@ def get_parameter(self, name: str) -> Parameter:
         return self._get_parameter(name=name, subtype=self._subtype)
 
 
-def _get_params_from_source(source: Callable) -> Optional[List[Parameter]]:
-    """Assembles an optional list of `Parameter` from the given `Callable` arguments.
+def _get_param_items_from_source(source: Callable) -> Optional[List[Parameter]]:
```

Comment:
I'm not sure the return needs to be Optional? An empty list suffices for existence check, and you can shortcut straight to
```py
for p in new_parameters:
```
at the call site